homework/index.js

```diff
@@ -30,17 +33,124 @@
     return elem;
   }
 
+  function createDetails(leftColumn, repositories) {
+    const table = createAndAppend('table', leftColumn, { id: 'main-table' });
+    const tBody = createAndAppend('tbody', table);
+    const repoDetails = ['Repository', 'Description', 'Forks', 'Updated'];
+    repoDetails.forEach(detail => {
+      const tr = createAndAppend('tr', tBody, { class: `${detail.toLowerCase()}-row` });
+      createAndAppend('td', tr, {
+        text: `${detail}: `,
+        class: `${detail.toLowerCase()}-header`,
+      });
+      createAndAppend('td', tr, { class: `${detail.toLowerCase()}-data` });
+    });
+    const repositoryName = document.querySelector('.repository-data');
+
+    // Functional requirement 4 -providing a link to the repository in the name property of the selected object
+
+    createAndAppend('a', repositoryName, {
+      href: repositories.html_url,
+      target: '_blank',
+      text: repositories.name,
+      class: 'repository-name',
+    });
+    if (repositories.description === null) {
+      document.querySelector(
+        '.description-data',
+      ).innerText = `No description is provided for this repository.`;
+    } else {
+      document.querySelector('.description-data').innerText = repositories.description;
+    }
+    document.querySelector('.forks-data').innerText = repositories.forks;
+    document.querySelector('.updated-data').innerText = new Date(
+      repositories.updated_at,
+    ).toLocaleString();
+  }
+
+  function createContributors(rightColumn, requestURL) {
+    fetchJSON(requestURL, (err, contributors) => {
+      if (err) {
```

Comment:
I would put the error message for getting contributors in the rightColumn rather than in root. If this query goes wrong, only the contributor data is missing, while the data over the repository can still show. As we don't know what the user is looking for, this repository data might be all they need (e.g. they want to see how many forks etc.).

Reply:
fixed this, I think I missed it due to my heavy use of ctrl + c and ctrl + v 😅